src/client/src/apps/StyleguideRoute/components/OnePageNavBar.tsx

```diff
@@ -20,49 +30,96 @@ const OnePageNavBar: React.FC<OnePageNavBar> = props => {
   const [scrollTop, setScrollTop] = useState(window.scrollY)
   const [positionNavBar, setPositionNavBar] = useState(MAX_SCROLL_HEIGHT)
   const [currentSection, setCurrentSection] = useState('')
-  const handleScroll = () => setScrollTop(window.scrollY)
+  const [isSticky, setIsSticky] = useState(false)
+
+  const handleScroll = useCallback(() => {
+    setScrollTop(window.scrollY)
+
+    const getCurrentSection = (): NavSection => {
```

Comment:
@fsarda there's now quite a lot going on in this event handler e.g creating a function & executing it. Inside that newly created function there's calculations too. The result for me is a fairly janky scroll performance. I also get a warning in the console:

<img width="596" alt="Screen Shot 2020-02-05 at 11 58 25" src="https://user-images.githubusercontent.com/156303/73840245-a4835080-480f-11ea-9bd0-3d7d7d9bcefc.png">

I'd suggest moving `getCurrentSection` outside of the scroll handler and wrapping it in a useCallback.

Reply:
Looking great @fsarda !